### PR TITLE
Explicit keywords

### PR DESCRIPTION
**Pull request by tjvr**
Some thoughts on #53.

See the [updated readme](https://github.com/tjvr/moo/blob/less-magic/README.md#keywords).

### CODE REVIEW

**Comment by tjvr** [comment]
I think making the keyword behaviour explicit is probably a good idea.

I'm not sure about this API, though. Perhaps it should be a nested object?
```js
moo.compile({
  identifier: {match: /[a-zA-Z]+/, keywords: {keyword: ['moo', 'cow']}},
})
```
Or with individual types:
```js
moo.compile({
  identifier: {
    match: /[a-zA-Z]+/,
    keywords: {
        'kw-class': 'class',
        'kw-def': 'def',
        'kw-if': 'if',
    },
  },
})
```

**Comment by nathan** [comment]
For the second form:

```js
itt(['class', 'def', 'if'])
.map(k => ['kw-' + k, k])
.toObject()
```

File: moo.js [hunk 1]
```diff
@@ -469,10 +433,55 @@
...
+    }
+
+    function str(x) { return JSON.stringify(x) }
+
+    var byLength = Object.create(null)
+    keywordList.forEach(function(keyword) {
+      (byLength[keyword.length] = byLength[keyword.length] || []).push(keyword)
+    })
+
+    // fast string lookup
```
Comment (nathan) [inline moo.js]: Have you profiled this as being faster than a plain old `switch` on the value?

Comment (tjvr) [inline moo.js]: Not personally, but [I've read](http://marijnhaverbeke.nl/blog/acorn.html) that switching on the length first is a helpful thing to do.

**Comment by tjvr** [comment]
> Iterator adaptors solve every* problem!

Nice! :-)

**Comment by tjvr** [comment]
> ```    keywords: {
>         'kw-class': 'class',
> ```

I'm also not convinced having the object "backwards" like this, for consistency with the outer `moo.compile`, is correct.

**Comment by tjvr** [comment]
Let's do this.

**Pull request merged**